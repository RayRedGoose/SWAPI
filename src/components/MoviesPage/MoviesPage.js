import React from 'react';
import UserProfile from '../UserProfile/UserProfile';
import MoviesBox from '../MoviesBox/MoviesBox';
import './MoviesPage.scss';

const MoviesPage = () => {

  return (
    <section className="moviesPage">
      <UserProfile />
      <h2>Star Wars Movies</h2>
      <MoviesBox />
    </section>
  )
}

export default MoviesPage;
