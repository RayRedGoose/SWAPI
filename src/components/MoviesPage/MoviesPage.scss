.moviePage {
  background: papayawhip;
}

.userProfile {
  display: block;
  position: relative;
  float: right;
  top: auto;
  border: 2px solid black;
  padding: 3em 5em 3em 5em;
    section {
      display: none;
    }
}

h2 {
  padding-top: 1em;
  padding-left: .5em;
  padding-bottom: 1.5em;
  font-size: 3.5em;
}
