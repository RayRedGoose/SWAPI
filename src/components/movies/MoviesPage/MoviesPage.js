import React from 'react';
import UserProfile from '../../UserProfile/UserProfile';
import MoviesBox from '../MoviesBox/MoviesBox';
import './MoviesPage.scss';

const MoviesPage = (props) => {
  return (
    <section className="moviePage">
      <header>
        <h2>Star Wars Movies</h2>
        <UserProfile
          {...props.user}
          logOut={props.logOut}
         />
      </header>
      <MoviesBox
        {...props.sys}
        movies={props.movies}
        addMovies={props.addMovies}
      />
    </section>
  )
}

export default MoviesPage;
