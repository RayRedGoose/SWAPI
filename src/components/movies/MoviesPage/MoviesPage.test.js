import React from 'react'
import { shallow } from 'enzyme'
import movieResults from '../../../mockData/fakeMovies.js'
import MoviesPage from './MoviesPage'

describe("MoviesPage", () => {

  it('should render MoviesPage with all data passed in correctly', () => {
    const wrapper = shallow(
      <MoviesPage
        sys={{
          isLoaded: true,
          error: ''
        }}
        user={{
          name: 'Padme',
          quote: "Rock is smooth",
          rank: 'expert'}}
        logOut={jest.fn()}
        movies={movieResults}
        addMovies={jest.fn()}
      />)

    expect(wrapper).toMatchSnapshot()
  })
})
