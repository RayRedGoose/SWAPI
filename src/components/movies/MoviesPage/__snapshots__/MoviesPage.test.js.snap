// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesPage should render MoviesPage with all data passed in correctly 1`] = `
<section
  className="moviePage"
>
  <Header
    headerText="Star Wars Movies"
    logOut={[MockFunction]}
    user={
      Object {
        "name": "Padme",
        "quote": "Rock is smooth",
        "rank": "expert",
      }
    }
  />
  <MovieBox
    addMovies={[MockFunction]}
    movies={
      Array [
        Object {
          "characters": Array [
            Object {
              "films": Array [
                "bambi",
                "little women",
                "sound of music",
              ],
              "isFavorite": false,
              "name": "Luke",
              "planet": "Tatioone",
              "population": 20000,
              "species": "human",
            },
            Object {
              "films": Array [
                "die hard",
              ],
              "isFavorite": true,
              "name": "Leia",
              "planet": "Alderaan",
              "population": 230000,
              "species": "human",
            },
            Object {
              "films": Array [
                "lawrence of arabia",
              ],
              "isFavorite": false,
              "name": "Obi-wan",
              "planet": "Whereever he is from",
              "population": 40000,
              "species": "human",
            },
            Object {
              "films": Array [
                "punch drunk love",
              ],
              "isFavorite": true,
              "name": "Darth Vadar",
              "planet": "Tatioone",
              "population": 50000,
              "species": "robot",
            },
            Object {
              "films": Array [
                "cinderella",
              ],
              "isFavorite": false,
              "name": "Han",
              "planet": "Millenium Falcon",
              "population": 60000,
              "species": "human",
            },
          ],
          "episode_id": 1,
          "opening_crawl": "paaaam-pam paaaaam-pam pa-pa-paaaam",
          "release_date": "1999-07-05",
          "title": "the fast and the furious",
        },
        Object {
          "characters": Array [
            Object {
              "films": Array [
                "bambi",
                "little women",
                "sound of music",
              ],
              "isFavorite": false,
              "name": "Luke",
              "planet": "Tatioone",
              "population": 20000,
              "species": "human",
            },
            Object {
              "films": Array [
                "die hard",
              ],
              "isFavorite": true,
              "name": "Leia",
              "planet": "Alderaan",
              "population": 230000,
              "species": "human",
            },
            Object {
              "films": Array [
                "lawrence of arabia",
              ],
              "isFavorite": false,
              "name": "Obi-wan",
              "planet": "Whereever he is from",
              "population": 40000,
              "species": "human",
            },
            Object {
              "films": Array [
                "punch drunk love",
              ],
              "isFavorite": true,
              "name": "Darth Vadar",
              "planet": "Tatioone",
              "population": 50000,
              "species": "robot",
            },
            Object {
              "films": Array [
                "cinderella",
              ],
              "isFavorite": false,
              "name": "Han",
              "planet": "Millenium Falcon",
              "population": 60000,
              "species": "human",
            },
          ],
          "episode_id": 2,
          "opening_crawl": "paaaam-pam paaaaam-pam pa-pa-paaaam",
          "release_date": "2000-07-05",
          "title": "2 fast 2 furious",
        },
        Object {
          "characters": Array [
            Object {
              "films": Array [
                "bambi",
                "little women",
                "sound of music",
              ],
              "isFavorite": false,
              "name": "Luke",
              "planet": "Tatioone",
              "population": 20000,
              "species": "human",
            },
            Object {
              "films": Array [
                "die hard",
              ],
              "isFavorite": true,
              "name": "Leia",
              "planet": "Alderaan",
              "population": 230000,
              "species": "human",
            },
            Object {
              "films": Array [
                "lawrence of arabia",
              ],
              "isFavorite": false,
              "name": "Obi-wan",
              "planet": "Whereever he is from",
              "population": 40000,
              "species": "human",
            },
            Object {
              "films": Array [
                "punch drunk love",
              ],
              "isFavorite": true,
              "name": "Darth Vadar",
              "planet": "Tatioone",
              "population": 50000,
              "species": "robot",
            },
            Object {
              "films": Array [
                "cinderella",
              ],
              "isFavorite": false,
              "name": "Han",
              "planet": "Millenium Falcon",
              "population": 60000,
              "species": "human",
            },
          ],
          "episode_id": 3,
          "opening_crawl": "paaaam-pam paaaaam-pam pa-pa-paaaam",
          "release_date": "2001-07-05",
          "title": "Tokyo Drift",
        },
        Object {
          "characters": Array [
            Object {
              "films": Array [
                "bambi",
                "little women",
                "sound of music",
              ],
              "isFavorite": false,
              "name": "Luke",
              "planet": "Tatioone",
              "population": 20000,
              "species": "human",
            },
            Object {
              "films": Array [
                "die hard",
              ],
              "isFavorite": true,
              "name": "Leia",
              "planet": "Alderaan",
              "population": 230000,
              "species": "human",
            },
            Object {
              "films": Array [
                "lawrence of arabia",
              ],
              "isFavorite": false,
              "name": "Obi-wan",
              "planet": "Whereever he is from",
              "population": 40000,
              "species": "human",
            },
            Object {
              "films": Array [
                "punch drunk love",
              ],
              "isFavorite": true,
              "name": "Darth Vadar",
              "planet": "Tatioone",
              "population": 50000,
              "species": "robot",
            },
            Object {
              "films": Array [
                "cinderella",
              ],
              "isFavorite": false,
              "name": "Han",
              "planet": "Millenium Falcon",
              "population": 60000,
              "species": "human",
            },
          ],
          "episode_id": 4,
          "opening_crawl": "paaaam-pam paaaaam-pam pa-pa-paaaam",
          "release_date": "2002-07-05",
          "title": "fast & furious",
        },
        Object {
          "characters": Array [
            Object {
              "films": Array [
                "bambi",
                "little women",
                "sound of music",
              ],
              "isFavorite": false,
              "name": "Luke",
              "planet": "Tatioone",
              "population": 20000,
              "species": "human",
            },
            Object {
              "films": Array [
                "die hard",
              ],
              "isFavorite": true,
              "name": "Leia",
              "planet": "Alderaan",
              "population": 230000,
              "species": "human",
            },
            Object {
              "films": Array [
                "lawrence of arabia",
              ],
              "isFavorite": false,
              "name": "Obi-wan",
              "planet": "Whereever he is from",
              "population": 40000,
              "species": "human",
            },
            Object {
              "films": Array [
                "punch drunk love",
              ],
              "isFavorite": true,
              "name": "Darth Vadar",
              "planet": "Tatioone",
              "population": 50000,
              "species": "robot",
            },
            Object {
              "films": Array [
                "cinderella",
              ],
              "isFavorite": false,
              "name": "Han",
              "planet": "Millenium Falcon",
              "population": 60000,
              "species": "human",
            },
          ],
          "episode_id": 5,
          "opening_crawl": "paaaam-pam paaaaam-pam pa-pa-paaaam",
          "release_date": "2003-07-05",
          "title": "fast 5",
        },
        Object {
          "characters": Array [
            Object {
              "films": Array [
                "bambi",
                "little women",
                "sound of music",
              ],
              "isFavorite": false,
              "name": "Luke",
              "planet": "Tatioone",
              "population": 20000,
              "species": "human",
            },
            Object {
              "films": Array [
                "die hard",
              ],
              "isFavorite": true,
              "name": "Leia",
              "planet": "Alderaan",
              "population": 230000,
              "species": "human",
            },
            Object {
              "films": Array [
                "lawrence of arabia",
              ],
              "isFavorite": false,
              "name": "Obi-wan",
              "planet": "Whereever he is from",
              "population": 40000,
              "species": "human",
            },
            Object {
              "films": Array [
                "punch drunk love",
              ],
              "isFavorite": true,
              "name": "Darth Vadar",
              "planet": "Tatioone",
              "population": 50000,
              "species": "robot",
            },
            Object {
              "films": Array [
                "cinderella",
              ],
              "isFavorite": false,
              "name": "Han",
              "planet": "Millenium Falcon",
              "population": 60000,
              "species": "human",
            },
          ],
          "episode_id": 6,
          "opening_crawl": "paaaam-pam paaaaam-pam pa-pa-paaaam",
          "release_date": "2004-07-05",
          "title": "furious 6",
        },
        Object {
          "characters": Array [
            Object {
              "films": Array [
                "bambi",
                "little women",
                "sound of music",
              ],
              "isFavorite": false,
              "name": "Luke",
              "planet": "Tatioone",
              "population": 20000,
              "species": "human",
            },
            Object {
              "films": Array [
                "die hard",
              ],
              "isFavorite": true,
              "name": "Leia",
              "planet": "Alderaan",
              "population": 230000,
              "species": "human",
            },
            Object {
              "films": Array [
                "lawrence of arabia",
              ],
              "isFavorite": false,
              "name": "Obi-wan",
              "planet": "Whereever he is from",
              "population": 40000,
              "species": "human",
            },
            Object {
              "films": Array [
                "punch drunk love",
              ],
              "isFavorite": true,
              "name": "Darth Vadar",
              "planet": "Tatioone",
              "population": 50000,
              "species": "robot",
            },
            Object {
              "films": Array [
                "cinderella",
              ],
              "isFavorite": false,
              "name": "Han",
              "planet": "Millenium Falcon",
              "population": 60000,
              "species": "human",
            },
          ],
          "episode_id": 7,
          "opening_crawl": "paaaam-pam paaaaam-pam pa-pa-paaaam",
          "release_date": "2005-07-05",
          "title": "furious 7",
        },
        Object {
          "characters": Array [
            Object {
              "films": Array [
                "bambi",
                "little women",
                "sound of music",
              ],
              "isFavorite": false,
              "name": "Luke",
              "planet": "Tatioone",
              "population": 20000,
              "species": "human",
            },
            Object {
              "films": Array [
                "die hard",
              ],
              "isFavorite": true,
              "name": "Leia",
              "planet": "Alderaan",
              "population": 230000,
              "species": "human",
            },
            Object {
              "films": Array [
                "lawrence of arabia",
              ],
              "isFavorite": false,
              "name": "Obi-wan",
              "planet": "Whereever he is from",
              "population": 40000,
              "species": "human",
            },
            Object {
              "films": Array [
                "punch drunk love",
              ],
              "isFavorite": true,
              "name": "Darth Vadar",
              "planet": "Tatioone",
              "population": 50000,
              "species": "robot",
            },
            Object {
              "films": Array [
                "cinderella",
              ],
              "isFavorite": false,
              "name": "Han",
              "planet": "Millenium Falcon",
              "population": 60000,
              "species": "human",
            },
          ],
          "episode_id": 8,
          "opening_crawl": "paaaam-pam paaaaam-pam pa-pa-paaaam",
          "release_date": "2006-07-05",
          "title": "the f8 of the furious",
        },
      ]
    }
  />
  <BBAnimated />
</section>
`;
